### PR TITLE
[MXNET-128] added load from buffer functions

### PR DESCRIPTION
**Pull request by dabraude**
## Description ##
Added load from function buffers for CPP package

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Added equivalent functions for the loading, from buffers instead of files 

## Comments ##
- Duplicates functionality from Python wrapper

### CODE REVIEW

**Comment by dabraude** [comment]
@cjolivier01 do you know who should be reviewing this change?

**Comment by cjolivier01** [comment]
anyone who choses to :)

File: cpp-package/include/mxnet-cpp/ndarray.hpp [hunk 1]
```diff
@@ -296,6 +296,57 @@ inline std::vector<NDArray> NDArray::LoadToList(const std::string &file_name) {
...
+                          std::map<std::string, NDArray> *array_map) {
+  mx_uint out_size, out_name_size;
+  NDArrayHandle *out_arr;
+  const char **out_names;
+  CHECK_EQ(MXNDArrayLoadFromBuffer(buffer, size, &out_size, &out_arr, &out_name_size,
+                         &out_names),
+           0);
+  if (array_list != nullptr) {
+    for (mx_uint i = 0; i < out_size; ++i) {
+      array_list->push_back(NDArray(out_arr[i]));
```
Comment (cjolivier01) [inline cpp-package/include/mxnet-cpp/ndarray.hpp]: please call array_list->reserve(out_size)

File: cpp-package/include/mxnet-cpp/ndarray.hpp [hunk 2]
```diff
@@ -296,6 +296,57 @@ inline std::vector<NDArray> NDArray::LoadToList(const std::string &file_name) {
...
+inline std::vector<NDArray> NDArray::LoadFromBufferToList(const void *buffer, size_t size) {
+  std::vector<NDArray> array_list;
+  mx_uint out_size, out_name_size;
+  NDArrayHandle *out_arr;
+  const char **out_names;
+  CHECK_EQ(MXNDArrayLoadFromBuffer(buffer, size, &out_size, &out_arr, &out_name_size,
+                         &out_names),
+           0);
+  for (mx_uint i = 0; i < out_size; ++i) {
+    array_list.push_back(NDArray(out_arr[i]));
```
Comment (cjolivier01) [inline cpp-package/include/mxnet-cpp/ndarray.hpp]: call reserve, please

**Comment by cjolivier01** [comment]
other than those two minor things, looks good

**Comment by dabraude** [comment]
Done the reserves, perhaps that should be put into the load from file functions too?

**Comment by dabraude** [comment]
balls need to redo this one

**Pull request closed without merging**